.github/workflows/main.yaml

```
@@ -0,0 +1,280 @@
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Pull  base image for proxy and Create Docker file
+      run: |
+          docker login -u $REGISTRY_USERNAME -p $REGISTRY_PASSWORD $SOURCE_IMAGE_REGISTRY
+          BASE_IMAGE=${SOURCE_IMAGE_REGISTRY}${SOURCE_IMAGE_NAME}:${SOURCE_IMAGE_TAG}
+          docker pull "${BASE_IMAGE}"
+          mkdir ${context} && cd ${context} 
+          echo "FROM ${BASE_IMAGE}" > Dockerfile
+          echo "RUN chown 1001:root /usr/bin/caddy" >> Dockerfile
+      if: contains(fromJSON('["aries-endorser-proxy"]'), matrix.service)
+      env:
+        context: ${{ matrix.SOURCE_CONTEXT_DIR }}
+        SOURCE_IMAGE_REGISTRY: "artifacts.developer.gov.bc.ca/docker-remote/"
+        SOURCE_IMAGE_NAME: caddy
+        SOURCE_IMAGE_TAG: latest
+        REGISTRY_USERNAME: ${{ secrets.USERNAME }}
+        REGISTRY_PASSWORD: ${{ secrets.PASSWORD }}
+
+    - name: Log in to the GHCR
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }} 
+
+    - name: Prepare docker tags for image
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: ghcr.io/bcgov/dts-endorser-service/${{ matrix.service }}
+        flavor: |
+          latest=true
+        tags: |
+          type=schedule
+          type=ref,event=branch
+          type=ref,event=pr
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=semver,pattern={{major}}
+          type=sha,value=latest
+      if: contains(fromJSON('["aries-endorser-agent","aries-endorser-backup","aries-endorser-api","aries-endorser-proxy"]'), matrix.service)
+      env: 
+        service: ${{ matrix.service }}
+    
+    - name: Build and push Docker image
+      id: docker_build
+      uses: docker/build-push-action@v5
+      with:
+        context: ${{ matrix.SOURCE_CONTEXT_DIR }}
+        file: ${{ matrix.SOURCE_CONTEXT_DIR }}/${{ matrix.DOCKER_FILE_PATH }}
+        push: true
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
+        cache-from: type=gha
+        cache-to: type=gha,mode=max
+      if: contains(fromJSON('["aries-endorser-agent","aries-endorser-backup","aries-endorser-api","aries-endorser-proxy"]'), matrix.service)
+
+    - name: Pull base image
+      id: pull_build_image
+      run: |
+          docker pull ${builder_image}
+      env:
+        builder_image: ${{ matrix.BUILDER_IMAGE }}
+      if: contains(fromJSON('["aries-endorser-db"]'), matrix.service)
+
+    - name: Setup and Build
+      id: build_image
+      uses: redhat-actions/s2i-build@v2
+      with:
+        path_context: ${{ matrix.PATH }}
+        # Builder image for a java project
+        builder_image: ${{ matrix.BUILDER_IMAGE }}
+        image:  dts-endorser-service/${{ matrix.service }}
+        tags: ${{ env.TAGS }}
```

Why not do this:
```suggestion
        tags: ${{ steps.meta.outputs.tags }}
```
and include aries-endorser-db in the Prepare docker tags for image step, and remove `env.TAGS`?

---

The steps.meta.outputs.tags is not in format required by OCP S2I build image, however i have created a step to conform steps.meta.outpus.tags to OCP S2I to build specification.